### ISSUE TITLE
trailing_blank_lines on empty, one-line file? Conflict with styler

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1445]
We have a few stub files for tests -- they need to match a filename pattern, but don't need any content.

The trouble is "how empty are they" causes some conflict with styler.

## File is just a newline -- `lintr` complains

```r
tmp <- tempfile(fileext = ".R")
# one-line empty file
cat("\n", file = tmp)
lintr::lint(tmp, lintr::trailing_blank_lines_linter())
# /tmp/Rtmpeek5O2/filed4a29466d762.R:1:1: style: [trailing_blank_lines_linter] Trailing blank lines are superfluous. Disable this with a comment: `# nolint` (one line) or `# lintr: disable` (multiple lines). See go/tidy-style#linting for more.
# 
# ^
```

## File is completely empty -- `styler` converts to newline

```r
file.remove(tmp)
file.create(tmp)
styler::style_file(tmp)
readChar(tmp, file.size(tmp))
# [1] "\n"
```

This causes a bit of friction as the file is never able to satisfy both `lintr` and `styler`. We could add a stub comment on the line but this shouldn't be necessary.

Filing in `lintr` for now, but please chime in @lorenzwalthert with your view on what should be done here.

I'm basicaly indifferent, but allowing `"\n"` to pass the `trailing_blank_lines_linter()` does feel right to me.

**Comment by lorenzwalthert** [comment]
I’d let this guide by:
- whether or not Unix touch command creates a new line or not upon file creation. 
- how it’s handled in other languages and linters/ formatters. 

I am fine to adapt {styler} to whatever.

**Comment by AshesITR** [comment]
IINM, `touch` creates a 0-byte empty file, not a 1-byte file containing `\n`.

**Comment by MichaelChirico** [comment]
```py
touch empty.py
flake8 empty.py
```

```sh
ls -l empty.py
# -rw-rw-r-- 1 michael michael 0 Jul  9 16:36 empty.py
```

Looks like `flake8` doesn't care about empty files, and `touch` creates a 0-byte file.

I'm not sure about the equivalent on Windows...

Maybe both packages should do a fix? `lintr` to ignore blank, 1-line files, and `styler` to leave empty files alone?

**Comment by lorenzwalthert** [comment]
What does flake8 and/or black do if a file contains just a newline?

**Comment by MichaelChirico** [comment]
Hmm, actually `flake8` complains about it too:

```
flake8 test.py 
# test.py:1:1: W391 blank line at end of file
```

**Comment by lorenzwalthert** [comment]
Thanks for the analysis. I suggest that only 0 bite files are allowed in both tools.

**Comment by AshesITR** [comment]
That means no todo for lintr:

``` r
tmp <- tempfile()
file.create(tmp)
#> [1] TRUE
length(lintr::lint(tmp))
#> [1] 0
```

<sup>Created on 2022-07-11 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by AshesITR** [comment]
Related: Is there a legitimate use for 0-byte chunks? If not, maybe lint them too?

**Comment by MichaelChirico** [comment]
> Related: Is there a legitimate use for 0-byte chunks? If not, maybe lint them too?

I am not 100% sure but I think the ones in our codebase are templates. In principle there can be chunk options (maybe custom ones) that induce side effects too, right?

**Comment by AshesITR** [comment]
Ah, yes, the `child` chunk arg is definitely a valid use case.
Over all, I think empty chunks should not contain blank lines to emphasize their speciality.
Using Ctrl + Alt + I in RStudio inserts an empty chunk with a blank line as a placeholder, so accidentally hitting the shortcut might produce stray empty chunks that would be found by a trailing_blank_lines_linter if we implement #741.

**Comment by IndrajeetPatil** [comment]
IINM, there is no longer a conflict between `{styler}` (dev version) and `{lintr}`.

## Before styling: 1 lint

```` 
tmp <- tempfile(fileext = ".Rmd")
cat(file = tmp, "
```{r chunk}

```
")

writeLines(readLines(tmp))
#> 
#> ```{r chunk}
#> 
#> ```

length(lintr::lint(tmp))
#> [1] 1
````

## After styling: 0 lints

````
styler::style_file(tmp)
#> Styling  1  files:
#>  /var/folders/xr/v_vddzvs33q5wg7jv9mr__4h0000gn/T//RtmpSLPiIc/file753230370e8a.Rmd ℹ 
#> ────────────────────────────────────────────────────────────────────────────────
#> Status   Count   Legend 
#> ✔    0   File unchanged.
#> ℹ    1   File changed.
#> ✖    0   Styling threw an error.
#> ────────────────────────────────────────────────────────────────────────────────
#> Please review the changes carefully!

writeLines(readLines(tmp))
#> 
#> ```{r chunk}
#> ```

length(lintr::lint(tmp))
#> [1] 0
````

<sup>Created on 2022-09-29 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
